### ISSUE TITLE
Domains: TXT record's value validation is broken

### ISSUE THREAD

**Issue by klimeryk** [Automattic/wp-calypso#3328]
* The value is being unnecessarily sanitized (lower case, strip whitespace)
* It's possible to submit an empty value (but the backend thankfully rejects that)

**Comment by andrewspittle** [comment]
I can confirm this happens in Calypso (and does not happen within the WP Admin Dashboard).

To reproduce:

1. Go to [the Domain Management screen](https://wordpress.com/domains/manage).
2. Select a WordPress.com site that has a domain mapped to it.
3. Select the mapped domain name.
4. Click "DNS Records"
5. Click "Add New DNS Record"
6. Select a TXT record.
7. Leave the "Name" field blank.
8. Enter a value in the "Text" field that includes a mix of uppercase and lowercase records. For example, [Google's example site verification value](https://support.google.com/a/answer/2716802): `google-site-verification=rXOxyZounnZasA8Z7oaD3c14JdjS9aKSWvsR1EbUSIQ`
9. Save the new TXT record.
10. The TXT record will be converted to lowercase.

When attempting the same TXT record addition through WP Admin (via Store > My Domains) the value saves correctly and preserves the upper and lowercase values.

This bug causes validation issues for services like Campaign Monitor that rely on TXT keys with mixed case values. [Example](https://twitter.com/galenking/status/703018679932293120).

**Comment by klimeryk** [comment]
Thanks for additional testing, @andrewspittle! Since this seems like an urgent issue (and I feel guilty, because it was me who introduced this regression :disappointed:), I've created a PR with a fix (#3585). Let me know if works for you :)

**Comment by BandonRandon** [comment]
Patch works for me
![screenshot_25-02-2016-21 30 16](https://cloud.githubusercontent.com/assets/150348/13344021/04c1a1dc-dc07-11e5-9109-5e910a79f2d7.png)

```
$ dig txt flagtest.me

;; ANSWER SECTION:
flagtest.me.   3600	IN	TXT	"google-site-verification=rXOxyZounnZasA8Z7oaD3c14JdjS9aKSWvsR1EbUSIQ"
```
